assets/js/modules/adsense/common/use-snippet-switch.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * AdSense Use Snippet Switch component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import Switch from '../../../components/switch';
+import SettingsNotice from '../../../components/settings-notice';
+import { trackEvent } from '../../../util';
+import { STORE_NAME } from '../datastore';
+const { useSelect, useDispatch } = Data;
+
+export default function UseSnippetSwitch( props ) {
+	const {
+		label = __( 'Let Site Kit place code on your site', 'google-site-kit' ),
+		checkedMessage,
+		uncheckedMessage,
+		saveOnChange,
+	} = props;
+
+	const useSnippet = useSelect( ( select ) => select( STORE_NAME ).getUseSnippet() );
+	const isDoingSaveUseSnippet = useSelect( ( select ) => select( STORE_NAME ).isDoingSaveUseSnippet() );
+
+	const { setUseSnippet, saveUseSnippet } = useDispatch( STORE_NAME );
+	const onChange = useCallback( async () => {
+		setUseSnippet( ! useSnippet );
+		trackEvent( 'adsense_setup', useSnippet ? 'adsense_tag_enabled' : 'adsense_tag_disabled' );
+		if ( saveOnChange ) {
```

Comment:
This is confusing—why wouldn't we want this setting to _always_  update? It's a better UX than having to click a save button somewhere. It might be nice to document why this is here in the code at least.

Reply:
This is only here for legacy reasons. We don't update settings immediately anywhere, and while this is generally a good practice and worth a consideration, we shouldn't add it at this point. The only reason this prop exists is that there is one place specifically in the AdSense setup where this value needs to update right away. Elsewhere (e.g. settings) it shouldn't.